### ISSUE TITLE
Tests aren't discovered if spec is a class

### ISSUE THREAD

**Issue by lukeknxt** [zio/zio#4493]
### Steps to reproduce
1. Create a class extending `DefaultRunnableSpec` and implement `spec`
2. Ensure test framework is setup in project settings
3. Try to run tests

### Result
- No tests are run
```
[info] No tests were executed
```

### Expected result
- Test suites are discovered

### Observation
- It seems that the spec has to be an `object` rather than a `class`. I was converting from ScalaTest so I got caught by this and it was not obvious via the docs as to what my problem could be.

### Potential solution
- Could we improve the docs around this or provide a way to discover classes and instantiate?

**Comment by hmemcpy** [comment]
The technical reason they have to be an `object` is that `DefaultRunnableSpec` is a `main` app - when you extend it, you get the ability to run it as if it were a regular command-line app, and with the [ZIO IntelliJ Plugin](https://plugins.jetbrains.com/plugin/13820-zio-for-intellij) you can run it straight from the IDE.

But I agree, this should probably be mentioned in the docs, although most (all?) test examples show an object.

**Comment by lukeknoot** [comment]
That makes sense. 

> But I agree, this should probably be mentioned in the docs, although most (all?) test examples show an object.

Yeah I guess for most people creating the tests from scratch it shouldn't be a problem. I already had a class so I suppose it wasn't what I was focusing on changing given everything else I had to do to go from ScalaTest -> ZIO testing.

A doc update would probably suffice if anything.

**Comment by jdegoes** [comment]
Yes, we should document the fact that all specs should be an object, not a class. ScalaTest, e.g., doesn't really care.
